patches/tModLoader/Terraria/NPC.cs.patch

```diff
@@ -885,7 +885,44 @@
  				Main.npc[nextNPC].GivenName = getNewNPCName(Type);
  				if (TownNPCProfiles.Instance.GetProfile(Type, out ITownNPCProfile profile)) {
  					Main.npc[nextNPC].townNpcVariationIndex = profile.RollVariation();
-@@ -61171,12 +_,14 @@
+@@ -61134,6 +_,7 @@
+ 		}
+ 
+ 		public int checkArmorPenetration(int armorPenetration) {
++			/*
+ 			if (armorPenetration <= 0)
+ 				return 0;
+ 
+@@ -61141,6 +_,17 @@
```

Comment:
Consider join these two patches by adding a comment here. 
Something like ``// Joined two adjacent patches``

Everything else looks reasonable in theory; removing armor penetration bonuses when defense is negative makes sense.

DefenseModifier as a frame-by-frame value also follows logically for handling debuffs. Wish there was a better way for it.

Reply:
I also vaguely wish there was a better way to do it, but this is the easiest way I can think of, and it lines up with how people expect these things to work as per player fields and existin' status effect logic, so it's probably fine